### PR TITLE
libideviceactivation: init at 1.1.1

### LINKED ISSUE #310787: Package request: libideviceactivation
**Project description**

Utility to activate iPhones

**Metadata**

* homepage URL: https://libimobiledevice.org
* source URL: https://github.com/libimobiledevice/libideviceactivation
* license: lgpl-2.1, GPL-3.0
* platforms: unix, linux, darwin, ...

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by clebs**
## Description of changes

Closes #310787

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/li/libideviceactivation/package.nix [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+  buildPhase = ''
+    runHook preBuild
+
+    ./autogen.sh
+    ./configure --prefix=""
+    make
+
+    runHook postBuild
+  '';
```
Comment (Gerg-L) [inline pkgs/by-name/li/libideviceactivation/package.nix:52]:
```suggestion
  installFlags = [ "PREFIX=$(out)" ];
```

File: pkgs/by-name/li/libideviceactivation/package.nix [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    owner = "libimobiledevice";
+    repo = "libideviceactivation";
+    rev = "refs/tags/${finalAttrs.version}";
+    hash = "sha256-owcQpCN4+A785oy9pCboJIyfpgZ6X+8PRzqGtWpYl2w=";
+  };
+
+  nativeBuildInputs = [
+    pkg-config
+    automake
+    autoconf
```
Comment (Gerg-L) [inline pkgs/by-name/li/libideviceactivation/package.nix:30]:
```suggestion
    autoreconfHook
```

File: pkgs/by-name/li/libideviceactivation/package.nix [hunk 3]
```diff
@@ -0,0 +1,65 @@
+{
+  stdenv,
+  fetchFromGitHub,
+  lib,
+  pkg-config,
+  automake,
+  autoconf,
```
Comment (Gerg-L) [inline pkgs/by-name/li/libideviceactivation/package.nix:7]:
```suggestion
  autoreconfHook,
```

**Review by Gerg-L — CHANGES_REQUESTED** [review]
Easier way to do it

File: pkgs/by-name/li/libideviceactivation/package.nix (hunk 2 above)
Comment (clebs) [reply pkgs/by-name/li/libideviceactivation/package.nix]: Thanks! This is really cool. I was always getting things stored in `$out/usr/local/bin | lib | share` but this takes care of everything.

PTAL!

**Comment by Gerg-L** [comment]
Result of `nixpkgs-review pr 314590` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>libideviceactivation</li>
  </ul>
</details>

**Review by Gerg-L — APPROVED** [review]

**Comment by tomberek** [comment]
Result of `nixpkgs-review pr 314590` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>libideviceactivation</li>
  </ul>
</details>

**Comment by Kreyren** [comment]
Thanks for packaging this! :heart:

**Pull request merged**